lego/apps/users/views/users.py

```diff
@@ -176,3 +176,35 @@ def change_grade(self, request, *args, **kwargs):
 
     def destroy(self, request, *args, **kwargs):
         return Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        permission_classes=[IsAuthenticated],
+        serializer_class=PhotoConsentSerializer,
+    )
+    def update_photo_consent(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = self.get_object()
+
+        with transaction.atomic():
+            if not request.user.has_perm(EDIT, user):
+                raise PermissionDenied(detail="Cannot update other user's consent")
+
+            photo_consent = None
+            try:
+                photo_consent = user.photo_consents.get(
+                    year=serializer.validated_data["year"],
+                    semester=serializer.validated_data["semester"],
+                    domain=serializer.validated_data["domain"],
+                )
+            except Exception:
+                pass
```

Comment:
eeeh what?
What is this meant to catch? And why are you just `pass`ing it?
If you need it, please catch the actual exception class and not the general one.

Reply:
Yeaaaah. That was a stupid fix for a test run. Removed altogether because I use update_or_create().